File: backend/app/schemas/workspace.py
"""
Схемы персонального workspace-хранилища.
"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


WORKSPACE_STORE_KEYS = frozenset({"company", "messenger", "support", "profile"})


class WorkspaceStoreRead(BaseModel):
    store_key: str
    data: dict[str, Any] | None = None
    updated_at: datetime | None = None


class WorkspaceStoreUpsert(BaseModel):
    data: dict[str, Any] = Field(default_factory=dict)

File: frontend/src/api/workspace.ts
import { apiJson } from './client'

export type WorkspaceStoreKey = 'company' | 'messenger' | 'support' | 'profile'

export type WorkspaceStore<T> = {
  store_key: WorkspaceStoreKey
  data: T | null
  updated_at: string | null
}

export function readWorkspaceStore<T>(
  token: string,
  storeKey: WorkspaceStoreKey,
): Promise<WorkspaceStore<T>> {
  return apiJson<WorkspaceStore<T>>(`/workspace/${storeKey}`, token)
}

export function saveWorkspaceStore<T extends Record<string, unknown>>(
  token: string,
  storeKey: WorkspaceStoreKey,
  data: T,
): Promise<WorkspaceStore<T>> {
  return apiJson<WorkspaceStore<T>>(`/workspace/${storeKey}`, token, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ data }),
  })
}

export function deleteWorkspaceStore(
  token: string,
  storeKey: WorkspaceStoreKey,
): Promise<void> {
  return apiJson<void>(`/workspace/${storeKey}`, token, {
    method: 'DELETE',
    parseJson: false,
  })
}
